### DISCUSSION TITLE
4D CT Analysis

### DISCUSSION THREAD

**Discussion by crcrewso** [jrkerns/pylinac#458] [Ideas]
CPQR's most recent Technical Quality Control Guidelines for CT Sims requires analyzing each phase annually for contrast, resolution, and slice thickness. [Page 11, Table 4](https://www.cpqr.ca/wp-content/uploads/2017/08/CTS-2016-07-02.pdf). For our clinic this would require manually exporting each phase individually and analyzing each using the CatPhan module. An alternate idea would be to use the DICOM header information to determine the phase, and then analyze each phase in turn, however this idea has a problem. 

As best as I can tell the only phase information being populated by our scanner is the Series Description 
```(0008, 103e) Series Description                  LO: '4D LUNG  2.0  Br38  3  60%'```

How does the group feel about manually parsing text fields that will be vendor, and possibly version dependent?

**Comment by jrkerns** [comment]
What phantom would this be analyzing?

Generally speaking, you could filter out the series based on the phases. E.g. here's a rudimentary filter:

```
from pylinac.core import DicomStack

stack = DicomImageStack(folder='/images/')
p60_stack = [img for img in stack.images if '60%' in img.SeriesDescription]
```

**Reply by crcrewso** [reply]
Thank you, that's helpful. I think we're going to use the CatPhan 604 on a custom sled, but I'd like to make the solution more generalizable.
